cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/source/DBSource.java

```diff
@@ -76,18 +80,19 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
                                   "authentication. If not, please remove dbUser and retry.");
     String jdbcPluginId = String.format("%s.%s.%s", "source", dbSourceConfig.jdbcPluginType,
                                         dbSourceConfig.jdbcPluginName);
-    Class<? extends Driver> jdbcDriverClass = pipelineConfigurer.usePluginClass(dbSourceConfig.jdbcPluginType,
+    driverClass = pipelineConfigurer.usePluginClass(dbSourceConfig.jdbcPluginType,
```

Comment:
Better for the `ensureValidConnection()` method takes a driver class as parameter. Also, you need to call `DBUtils.cleanup(driverClass);` with a finally block. The destroy method will not be called during configure phase.